apps/aehttp/src/aehttp_dispatch_int.erl

```diff
@@ -15,328 +34,399 @@
+                 base58_decode([{channel_id, channel_id, {id_hash, [channel]}},
+                                {from_id, from_id, {id_hash, [account_pubkey]}},
+                                {poi, poi, poi}]),
+                 get_nonce_from_account_id(from_id),
+                 poi_decode(poi),
+                 unsigned_tx_response(fun aesc_slash_tx:new/1)
+                ],
+    process_request(ParseFuns, Req);
+
+handle_request('PostChannelSettle', #{'ChannelSettleTx' := Req}, _Context) ->
+    ParseFuns = [parse_map_to_atom_keys(),
+                 read_required_params([channel_id, from_id,
+                                       initiator_amount_final,
+                                       responder_amount_final,
+                                       fee, nonce]),
+                 read_optional_params([{ttl, ttl, '$no_value'}]),
+                 base58_decode([{channel_id, channel_id, {id_hash, [channel]}},
+                                {from_id, from_id, {id_hash, [account_pubkey]}}]),
+                 unsigned_tx_response(fun aesc_settle_tx:new/1)
+                ],
+    process_request(ParseFuns, Req);
+
+handle_request('PostOracleRegister', #{'OracleRegisterTx' := Req}, _Context) ->
+    ParseFuns = [parse_map_to_atom_keys(),
+                 read_required_params([account_id, {query_format, query_format},
+                                       {response_format, response_format},
+                                       query_fee, oracle_ttl, fee]),
+                 read_optional_params([{ttl, ttl, '$no_value'}]),
+                 base58_decode([{account_id, account_id, {id_hash, [account_pubkey]}}]),
+                 get_nonce_from_account_id(account_id),
+                 ttl_decode(oracle_ttl),
+                 unsigned_tx_response(fun aeo_register_tx:new/1)
+                ],
+    process_request(ParseFuns, Req);
+
+handle_request('PostOracleExtend', #{'OracleExtendTx' := Req}, _Context) ->
+    ParseFuns = [parse_map_to_atom_keys(),
+                 read_required_params([oracle_id, oracle_ttl, fee]),
+                 read_optional_params([{ttl, ttl, '$no_value'}]),
+                 base58_decode([{oracle_id, oracle_id, {id_hash, [oracle_pubkey]}}]),
+                 get_nonce_from_account_id(oracle_id),
+                 ttl_decode(oracle_ttl),
+                 unsigned_tx_response(fun aeo_extend_tx:new/1)
+                ],
+    process_request(ParseFuns, Req);
+
+handle_request('PostOracleQuery', #{'OracleQueryTx' := Req}, _Context) ->
+    ParseFuns = [parse_map_to_atom_keys(),
+                 read_required_params([sender_id, oracle_id, query,
+                                       query_fee, fee, query_ttl, response_ttl]),
+                 read_optional_params([{ttl, ttl, '$no_value'}]),
+                 base58_decode([{sender_id, sender_id, {id_hash, [account_pubkey]}},
+                                {oracle_id, oracle_id, {id_hash, [oracle_pubkey]}}]),
```

Comment:
I see this means recipient of query only oracle pubkey - e.g. not name. cc @sennui @zp-sd

Reply:
That is in line with https://github.com/aeternity/protocol/blame/f622f21f6b41fb5860fd9f37e8df2d5ea7bdb705/oracles/oracle_transactions.md#L68 so looks ok.